### PR TITLE
fix(aws)!: Missing partial/parent PK for `aws_ssm_instance_patches`

### PR DESCRIPTION
**Pull request by disq**
Also clean up SSM instances code structure

### CODE REVIEW

File: plugins/source/aws/resources/services/ssm/instance_compliance_items.go [hunk 1]
```diff
@@ -28,11 +32,33 @@ func InstanceComplianceItems() *schema.Table {
 			{
 				Name:     "instance_arn",
 				Type:     schema.TypeString,
-				Resolver: resolveInstanceComplianceItemInstanceARN,
+				Resolver: schema.ParentColumnResolver("arn"),
```
Comment (disq) [inline plugins/source/aws/resources/services/ssm/instance_compliance_items.go:35]: Change 2: Don't re-build ARN when the parent already has the same value

File: plugins/source/aws/resources/services/ssm/instance_patches.go [hunk 2]
```diff
@@ -15,8 +18,16 @@ func InstancePatches() *schema.Table {
...
 		Multiplex:   client.ServiceAccountRegionMultiplexer("ssm"),
 		Columns: []schema.Column{
-			client.DefaultAccountIDColumn(true),
-			client.DefaultRegionColumn(true),
+			client.DefaultAccountIDColumn(false),
+			client.DefaultRegionColumn(false),
+			{
+				Name:     "instance_arn",
+				Type:     schema.TypeString,
+				Resolver: schema.ParentColumnResolver("arn"),
```
Comment (disq) [inline plugins/source/aws/resources/services/ssm/instance_patches.go:26]: Change 1: Add parent `arn` here as PK, drop account_id/region PKs

File: plugins/source/aws/resources/services/ssm/instance_compliance_items_fetch.go [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-	"fmt"
-
-	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/aws/arn"
-	"github.com/aws/aws-sdk-go-v2/service/ssm/types"
-	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
-	"github.com/cloudquery/plugin-sdk/schema"
-)
-
-func resolveInstanceComplianceItemInstanceARN(_ context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```
Comment (disq) [inline plugins/source/aws/resources/services/ssm/instance_compliance_items_fetch.go:14]: Part of change 2.

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Looks good, but since it needs to be a breaking change let's hold off on it for a bit to give us time to explore ways to make these transitions easier on users (https://github.com/cloudquery/cloudquery/issues/8474)

**Pull request merged**